lib/validator/v3/validator.go

```diff
@@ -427,6 +429,49 @@ func validateCIDR(v *validator.Validate, topStruct reflect.Value, currentStructO
 	return err == nil
 }
 
+// validateHTTPMethods validate if the HTTP Method match clauses are valid.
+func validateHTTPMethods(v *validator.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
```

Comment:
These functions need to use structlevel.ReportError to tell the user what is wrong.